### PR TITLE
Move protoc artefact to version catalog

### PR DESCRIPTION
**Pull request by oas004**
#### WHAT
Move protoc artefact to version catalog

#### WHY
relates to #1954 

#### HOW
Move out dependencies and refactor

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [ ] Run tests
- [x] Update metalava's signature text files

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -193,6 +193,7 @@ playservices-auth = { module = "com.google.android.gms:play-services-auth", vers
 playservices-base = "com.google.android.gms:play-services-base:18.5.0"
 playservices-wearable = "com.google.android.gms:play-services-wearable:18.2.0"
 protobuf-kotlin-lite = { module = "com.google.protobuf:protobuf-kotlin-lite", version.ref = "protobuf" }
+protobuf-proto = { module = "com.google.protobuf:protoc", version.ref = "protobuf" }
```
Comment (yschimke) [inline gradle/libs.versions.toml:196]: Should be protoc?

```suggestion
protobuf-protoc = { module = "com.google.protobuf:protoc", version.ref = "protobuf" }
```

File: datalayer/core/build.gradle.kts [hunk 2]
```diff
@@ -77,7 +77,7 @@ android {
 
 protobuf {
     protoc {
-        artifact = "com.google.protobuf:protoc:4.28.1"
+        artifact = libs.protobuf.proto.get().toString()
```
Comment (yschimke) [inline datalayer/core/build.gradle.kts:80]: Can we also move protoc-gen-javalite?

File: gradle/libs.versions.toml (hunk 1 above)
Comment (oas004) [reply gradle/libs.versions.toml:196]: oh ups! Thank you for catching that! 🙏

File: datalayer/core/build.gradle.kts (hunk 2 above)
Comment (oas004) [reply datalayer/core/build.gradle.kts:80]: Yes good idea! Is it preferable to do it in this PR or would you want me to separate it? :)

Comment (oas004) [reply datalayer/core/build.gradle.kts]: I just added it in this :) Hope it was okay :)

**Review by yschimke — APPROVED** [review]

**Pull request merged**